akka-projection-core/src/main/java/akka/projection/internal/spi/ProjectionMetrics.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.internal.spi;
+
+import akka.actor.ClassicActorSystemProvider;
+
+public interface ProjectionMetrics {
+
+    /**
+     * Must be invoked for each element received from the Source when the Source provides it. If possible,
+     * invoke this method in a stream stage as close to the stream stage where you read from the wire to
+     * measure the parsing, deserializing and other steps of the processing prior to the event handling.
+     *
+     * @param projectionName the projection name
+     * @param projectionKey  the unique key for this Projection instance
+     * @param creationTimestamp if the element traversing the stream contains the creation time, provide it. Set
```

Comment:
A `ProjectionId` that combines name and key will be merged as part of the slick implementation soon #35.

Reply:
Yeah, I'm counting on that. ;-)